junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/extension/TempDirectory.java

```diff
@@ -250,6 +281,19 @@ public void close() throws IOException {
 			if (!failures.isEmpty()) {
 				throw createIOExceptionWithAttachedFailures(failures);
 			}
+
+			closeFactory();
+		}
+
+		private void closeFactory() throws IOException {
+			try {
+				if (factory instanceof AutoCloseable) {
+					((AutoCloseable) factory).close();
+				}
```

Comment:
Would it make sense for `TempDirFactory` to extend `AutoCloseable` with an empty default implementation of `close()`?

Reply:
Triggered by the integration tests added in 24b6a0b5, I went ahead with this change. Please, let me know if you see anything wrong with it.